### DISCUSSION TITLE
Download/Upload Buffer Purpose

### DISCUSSION THREAD

**Discussion by mepherion** [tnodir/fort#214] [Q&A] [answered]
What does the download and upload buffer setting do? When would you need to change it from the default values?

**Comment by tnodir** [comment, marked as answer]
Most of the code for speed limiting was taken from the [win-shaper](https://github.com/catchpoint/WebPageTest.win-shaper).

Please see its [blog post](http://calendar.perfplanet.com/2016/testing-with-realistic-networking-conditions/):

> By default it ships with the same connection profiles as WebPageTest but you can also specify custom profiles including how big the network buffers should be (150KB roughly mirrors the dummynet default of 100 packets though in practice buffers on real networks can be even larger).

**Comment by tnodir** [comment]
Let's keep this open for documentation.
